system/src/system_power_manager.cpp

```diff
@@ -142,9 +142,14 @@ void PowerManager::sleep(bool s) {
       // Wake up
       initDefault();
       update();
-      gauge.wakeup();
-      // Delay for at least 500ms to make sure the fuelgauge is ready to function after woken up.
-      HAL_Delay_Milliseconds(500);
+      if (isRunning()) {
+        Event ev = Event::Wakeup;
+        os_queue_put(queue_, (const void*)&ev, CONCURRENT_WAIT_FOREVER, nullptr);
```

Comment:
This should be done before or combined with `update()` call. As otherwise the fuelgauge will anyway be accessed in `handleUpdate()`.